blockdag/blocknode_test.go

```diff
@@ -109,24 +117,27 @@ func TestBlueAnticoneSizesSize(t *testing.T) {
 	}
 	defer teardownFunc()
 	blockHeader := dagconfig.SimnetParams.GenesisBlock.Header
-	block, _ := dag.newBlockNode(&blockHeader, newSet())
+	node, _ := dag.newBlockNode(&blockHeader, newSet())
+	hash := daghash.Hash{1}
 
-	maxKsize := dagconfig.KType(math.MaxUint8)
-	expected := maxKsize
-	block.bluesAnticoneSizes[daghash.Hash{1}] = expected
-	serializedNode, _ := serializeBlockNode(block)
+	// Setting maxKType to maximum value fo KType
+	maxKType := ^dagconfig.KType(0)
+	node.bluesAnticoneSizes[hash] = maxKType
+	serializedNode, _ := serializeBlockNode(node)
 	deserializedNode, _ := dag.deserializeBlockNode(serializedNode)
-	if deserializedNode.bluesAnticoneSizes[daghash.Hash{1}] != expected {
+	if deserializedNode.bluesAnticoneSizes[hash] != maxKType {
 		t.Fatalf("TestBlueAnticoneSizesSize: BlueAnticoneSize should not change when deserilizing. Expected: %v but got %v",
-			expected, deserializedNode.bluesAnticoneSizes[daghash.Hash{1}])
+			maxKType, deserializedNode.bluesAnticoneSizes[hash])
 	}
 
-	block.bluesAnticoneSizes[daghash.Hash{1}]++
-	size := block.bluesAnticoneSizes[daghash.Hash{1}]
-	serializedNode, _ = serializeBlockNode(block)
+	// Increasing bluesAnticoneSizes by 1 to make it overflow
```

Comment:
I think this part of the test will always pass, because 0 is always serializable. I think that the first part might suffice

Reply:
I agree that this test right now will always pass, but I'd argue the first part will not suffice.
The first part will catch the case where bluesAnticoneSizes is smaller than KType, but if it's wider it will work just fine.

I think in second part we should set `expected := 0` (Or `maxKType + 1`, which equals 0), and than it will check what we want to check.